experimental/segmenter/data/word.json

```diff
@@ -0,0 +1,148 @@
+{
+"segmenter_type": "word",
```

Comment:
Some high level questions about the data.

1. Is `word.json' crafted by hand?
2. How "rules" are generated from https://www.unicode.org/reports/tr29/#Word_Boundary_Rules? I feel we might need some comment to associate each rule to the spec text. For example I guess line 125 `{ "left": ["sot"], "right": ["Any"], "break_state": true },` is correspond to [WB1](https://www.unicode.org/reports/tr29/#WB1). However, other more complex rule might not be obvious. Could we improve the readability?
3. We might need to encode Unicode version number in the json. For example, if the any future Unicode release tweaks the rule, we'll know what to compare.

Reply:
This file is by hand. I don't add any comment due to json file. Should I add anything comment fields although it isn't parsed and used?

Reply:
Ah, I see. JSON format doesn't allow comment syntax like `//` or `/* .. */` . 

If this file is generated manually, adding comment fields to workaround JSON is probably not the optimal solution. It is better to use a human friendly format like TOML. Otherwise this file would become a maintenance burden if it has no comment, or the comment fields are mixed with the data.

I see @sffc has code to deserialize TOML in https://github.com/unicode-org/icu4x/blob/c7db0802d44dd9c79518a29712f63cbb8b2159d7/provider/uprops/src/uprops_helpers.rs#L32-L34  Technically, It should be possible to use TOML instead of JSON, and we can use regular comment syntax `#`. What do you think?